### PR TITLE
fix(semver): throw on invalid input in `parseRange()`

### LINKED ISSUE #4845: semver: `parseRange` doesn't throw with invalid input
The below example doesn't throw a type error:

```ts
import { parseRange } from "@std/semver";
const range = parseRange("invalid");
console.log(range);
```

Instead it prints the below range object:
```js
[
  [
    {
      operator: "<",
      major: 0,
      minor: 0,
      patch: 0,
      prerelease: [],
      build: []
    }
  ]
]
```

### PR DESCRIPTION
**Pull request by iuioiua**
Closes #4845

### CODE REVIEW

File: semver/is_range.ts [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
+import { ALL } from "./_constants.ts";
 import { isSemVer } from "./is_semver.ts";
 
 function isComparator(value: unknown): value is Comparator {
   if (
     value === null || value === undefined || Array.isArray(value) ||
     typeof value !== "object"
   ) return false;
-  if (value === NONE || value === ALL) return true;
+  if (value === ALL) return true;
```
Comment (iuioiua) [inline semver/is_range.ts:13]: Here, we no longer regard `<0.0.0` as a valid comparator.

File: semver/parse_range.ts [hunk 2]
```diff
@@ -404,9 +404,13 @@ function parseOperatorRanges(string: string): Comparator[] {
  * @returns A valid SemVer range
  */
 export function parseRange(range: string): Range {
-  return range
+  const result = range
     // remove spaces between operators and versions
-    .replaceAll(/(?<=<|>|=|~) +/g, "")
+    .replaceAll(/(?<=<|>|=|~)(\s+)/g, "")
```
Comment (iuioiua) [inline semver/parse_range.ts:409]: Pass-by fix that this change requires.

File: semver/parse_range.ts [hunk 3]
```diff
@@ -404,9 +404,13 @@ function parseOperatorRanges(string: string): Comparator[] {
  * @returns A valid SemVer range
  */
 export function parseRange(range: string): Range {
-  return range
+  const result = range
     // remove spaces between operators and versions
-    .replaceAll(/(?<=<|>|=|~) +/g, "")
+    .replaceAll(/(?<=<|>|=|~|^)(\s+)/g, "")
```
Comment (iuioiua) [inline semver/parse_range.ts:409]: 1. Addresses https://github.com/denoland/std/pull/5564#discussion_r1694506670
2. Required fix for this PR.

Comment (kt3k) [reply semver/parse_range.ts:409]: Can you also add a test case for checking `^ 1.0.0` works?

File: semver/parse_range.ts [hunk 4]
```diff
@@ -404,9 +404,13 @@ function parseOperatorRanges(string: string): Comparator[] {
...
     .split(/\s*\|\|\s*/)
     .map((string) => parseHyphenRange(string) || parseOperatorRanges(string));
+  if (result.some((r) => r.includes(null))) {
+    throw new TypeError(`Invalid range: ${range}`);
+  }
```
Comment (kt3k) [inline semver/parse_range.ts:414]: nice 👍

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**